README.md

```diff
@@ -23,396 +23,373 @@ Stable release versions are manually released from the main branch.
-unbondingTx.setWitness(0, witness);
+const { 
+  signedBabylonTx
+} = await manager.preStakeRegistrationBabylonTransaction(
+  stakerInfo,
+  stakingInput,
+  babylonBtcTipHeight,
+  inputUTXOs,
+  feeRate,
+  bech32Address,
+);
 ```
 
-### Withdrawing
+### Staking Transaction
 
-Withdrawing involves extracting funds for which the staking/unbonding period
-has expired from the staking/unbonding transaction.
+This step signs the previously created (but unsigned) Bitcoain staking transaction, 
+making it ready for submission to the Bitcoin network. The hash of the signed 
+staking transaction must match the staking transaction hash contained within 
+the pre-staking registration Babylon transaction.
 
-Initially, we specify the withdrawal transaction parameters.
+The unsigned staking transaction can be retrieved from either the Babylon
+chain or the backend API.
 
 ```ts
-// The index of the staking/unbonding output in the staking/unbonding
-// transcation.
-const stakingOutputIndex: number = 0;
-
-// The fee that the withdrawl transaction should use.
-const withdrawalFee: number = 500;
+import {
+  BabylonBtcStakingManager,
+} from "@babylonlabs-io/btc-staking-ts";
+
+const manager = new BabylonBtcStakingManager(
+  btcNetwork,
+  stakingParams,
+  btcProvider,
+  bbnProvider,
+);
 
-// The address to which the funds should be withdrawed to.
-const withdrawalAddress: string = btcWallet.address;
+const signedBtcStakingTx = await manager.createSignedBtcStakingTransaction({
+  stakerInfo,
+  stakingInput,
+  unsignedStakingTx,
+  inputUTXOs,
+  stakingParamsVersion
+})
 ```
 
-Then, we construct the withdrawal transaction.
-There are two types of withdrawal
+### Unbonding Transaction
 
-1. Withdraw funds from a staking transaction in which the timelock naturally
-   expired:
+This step signs the previously created unbonding transaction and 
+combines it with the signatures from covenants to create a properly signed 
+unbonding transaction that is ready to be sent to the Bitcoin network to unbond 
+a staking transaction.
 
-```ts
-import { Psbt, Transaction } from "bitcoinjs-lib";
-import { withdrawTimelockUnbondedTransaction } from "@babylonlabs-io/btc-staking-ts";
-
-// staking transaction. Transaction
-const stakingTx: Transaction = undefined;
-
-const unsignedWithdrawalPsbt: {psbt: Psbt, fee: number} = withdrawTimelockUnbondedTransaction(
-  scripts: {
-    timelockScript,
-    slashingScript,
-    unbondingScript,
-  },
-  stakingTx,
-  btcWallet.address,
-  network,
-  feeRate,
-  stakingOutputIndex,
-);
-```
-
-2. Withdraw funds from an unbonding transaction that was submitted for early
-   unbonding and the unbonding period has passed:
+The unsigned unbonding and staking transactions, 
+as well as the covenant committee signatures, can be retrieved from either the 
+Babylon Genesis Chain or backend API using the endpoints described 
+in [Staking Transaction Details](#1-staking-transaction-details).
```

Comment:
We can list their retrieval here, similar as to how I referenced the retrieval of the network parameters previously.